### PR TITLE
Make snaps-cli windows compatible

### LINKED ISSUE #83: Make compatible with Windows
`snaps-cli` should be compatible with Windows. This is mostly of ensuring that it can handle both the Windows and *nix filesystems. Currently, paths are hardcoded to assume a Unix platform.

### PR DESCRIPTION
**Pull request by hmalik88**
Closes #83

Since *nix and windows systems have different syntax for paths, we need to make our cli os agnostic. This is done through utilizing the `path` module. 

- [x] Add `path.normalize` to `sanitizeInputs` in `misc.ts`
- [x] Normalize hardcoded paths
- [x] Get cli to run on windows
- [x] Update tests

### CODE REVIEW

**Review by rekmarks — CHANGES_REQUESTED** [review]
Looks good! See inline change requests. The biggest one is to use `path.normalize` instead of path literals and conditions using `os.platform()` in tests. Please let me know if we're intentionally not using `path.normalize`!

File: packages/snaps-cli/scripts/createInitTemplate.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const EXAMPLE_PATH = path.join(
   __dirname,
   '../../../node_modules/@metamask/snap-examples/examples/hello-snaps',
 );
-const TEMPLATE_PATH = 'src/cmds/init/init-template.json';
+const TEMPLATE_PATH = path.normalize('src/cmds/init/init-template.json');
```
[the remark this answers is not in the document]
Comment (hmalik88) [reply packages/snaps-cli/scripts/createInitTemplate.js:8]: `TEMPLATE_PATH` is only passed into the `fs.writeFile` if I'm not mistaken. But you're right nonetheless, `fs.*` methods can accept unix OR windows paths. Going to change this.

File: packages/snaps-cli/src/utils/misc.ts [hunk 2]
```diff
@@ -13,6 +14,17 @@ export const permRequestKeys = [
 
 export const CONFIG_PATHS = ['snap.config.json'];
 
+const argumentSet = new Set([
```
[the remark this answers is not in the document]
Comment (hmalik88) [reply packages/snaps-cli/src/utils/misc.ts:17]: Will do, thanks!

File: packages/snaps-cli/src/cmds/build/build.test.ts [hunk 3]
```diff
@@ -21,7 +22,11 @@ describe('build', () => {
         manifest: true,
       };
 
-      const outfilePath = `${mockArgv.dist}/${mockArgv.outfileName}`;
+      const outfilePath =
```
[the remark this answers is not in the document]
Comment (hmalik88) [reply packages/snaps-cli/src/cmds/build/build.test.ts:25]: Ahh 😅, I guess I was trying to be explicit in a sense with the paths, but yeah `path.normalize()` does the same thing -- will change!

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**